pkgs/applications/science/math/sage/sage-env.nix

```diff
@@ -0,0 +1,158 @@
+{ stdenv
+, lib
+, writeTextFile
+, python
+, sage-src
+, sagelib
+, env-locations
+, gfortran
+, bash
+, coreutils
+, gnused
+, gnugrep
+, binutils
+, pythonEnv
+, python3
+, pkg-config
+, pari
+, gap-libgap-compatible
+, libgap
+, ecl
+, maxima-ecl
+, singular
+, giac
+, palp
+, rWrapper
+, gfan
+, cddlib
+, jmol
+, tachyon
+, glpk
+, eclib
+, sympow
+, nauty
+, sqlite
+, ppl
+, ecm
+, lcalc
+, rubiks
+, flintqs
+, openblas-cblas-pc
+, flint
+, gmp
+, mpfr
+, pynac
+, zlib
+, gsl
+, ntl
+}:
+
+let
+  runtimepath = (lib.makeBinPath ([
+    "@sage-local@"
+    "@sage-local@/build"
+    pythonEnv
+    python3.buildEnv # empty env to clear PYTHONHOME
```

Comment:
I don't follow this. Adding just `python3` is not sufficient?

Reply:
No, because then `wrapper.nix` won't be used and `PYTHONHOME` won't be re-set, see
https://github.com/NixOS/nixpkgs/blob/57bccb3cb8f6b5dea5c315fcaabdb992498b9314/pkgs/development/interpreters/python/wrapper.nix#L35

If I recall correctly that is important because sage calls the python3 binary from within python2. That means the python2 `PYTHONHOME` will be set and confuse python3.

The comment isn't very helpful though, I'll add more clarification.